lib/feature/add_character/add_character_page.dart

```diff
@@ -0,0 +1,204 @@
+import 'package:character_viewer/common/common.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import 'cubit/add_character_cubit.dart';
+
+class AddCharacterPage extends StatelessWidget {
+  const AddCharacterPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => getIt<AddCharacterCubit>(),
+      child: Scaffold(
+        appBar: AppBar(
+          title: const Text('Add Character'),
+        ),
+        body: _AddCharacterView(),
+      ),
+    );
+  }
+}
+
+class _AddCharacterView extends StatelessWidget {
+  final TextEditingController titleController = TextEditingController();
+  final TextEditingController imageUrlController = TextEditingController();
+  final TextEditingController descriptionController = TextEditingController();
+
+  _AddCharacterView({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<AddCharacterCubit, AddCharacterState>(
+      listener: (context, state) {
+        if (state is AddCharacterStateSuccess) {
+          ScaffoldMessenger.of(context).showSnackBar(
+            SnackBar(
+              content: Text('Character added: ${state.newCharacter.title}'),
+              backgroundColor: Colors.green,
+            ),
+          );
+        }
+      },
+      builder: (context, state) {
+        return Scaffold(
+          body: _buildUI(context, state),
+        );
+      },
+    );
+  }
+
+  Widget _buildUI(BuildContext context, AddCharacterState state) {
+    if (state is AddCharacterStateInitial) {
+      return _buildInitialUI(context);
+    } else if (state is AddCharacterStateSuccess) {
+      return _buildSuccessUI(context, state.newCharacter);
+    } else if (state is AddCharacterStateFailure) {
+      return _buildErrorUI(context, state.error);
+    } else {
+      return const Center(
+        child: CircularProgressIndicator(),
+      );
+    }
+  }
+
+  Widget _buildInitialUI(BuildContext context) {
```

Comment:
change helper methods to widgets

https://www.youtube.com/watch?v=IOyq-eTRhvo&ab_channel=Flutter

Reply:
Done